### ISSUE TITLE
ModuleNotFoundError for pandas when hosted in Azure

### ISSUE THREAD

**Issue by elliotchaim** [Azure/azure-functions-python-worker#464]
Python function, works fine locally using either VS Code and command line. Deployed using VS Code. Expected to work fine in Azure. However, In Azure I get a 500 response. Error code is 'ModuleNotFoundError'. Same result on both Consumption and Dedicated plans. Full error message below:

```
Result: Failure Exception: ModuleNotFoundError: No module named 'pandas' Stack: File "/usr/local/lib/python3.6/site-packages/azure/functions_worker/dispatcher.py", line 230, in _handle__function_load_request func_request.metadata.entry_point) File "/usr/local/lib/python3.6/site-packages/azure/functions_worker/loader.py", line 66, in load_function mod = importlib.import_module(fullmodname) File "/usr/local/lib/python3.6/importlib/__init__.py", line 126, in import_module return _bootstrap._gcd_import(name[level:], package, level) File "<frozen importlib._bootstrap>", line 994, in _gcd_import File "<frozen importlib._bootstrap>", line 971, in _find_and_load File "<frozen importlib._bootstrap>", line 941, in _find_and_load_unlocked File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed File "<frozen importlib._bootstrap>", line 994, in _gcd_import File "<frozen importlib._bootstrap>", line 971, in _find_and_load File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked File "<frozen importlib._bootstrap>", line 665, in _load_unlocked File "<frozen importlib._bootstrap_external>", line 678, in exec_module File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed File "/home/site/wwwroot/Predict/__init__.py", line 6, in <module> from .SharedCode.predict import predict File "/home/site/wwwroot/Predict/SharedCode/predict.py", line 4, in <module> import pandas as pd
```

There are other modules in the same file which don't throw this error. See head of predict.py:

```
import logging
import pickle
from . import (utils, data)
import pandas as pd

class predict():
...
```

My requirements.txt file:

```txt
astroid==2.2.5
azure-functions==1.0.0b4
colorama==0.4.1
isort==4.3.20
joblib==0.13.2
lazy-object-proxy==1.4.1
mccabe==0.6.1
numpy==1.16.3
pandas==0.24.2
pylint==2.3.1
python-dateutil==2.8.0
pytz==2019.1
scikit-learn==0.21.1
scipy==1.3.0
six==1.12.0
typed-ast==1.3.5
wrapt==1.11.1
```

**Comment by Hazhzeng** [comment]
Hello @elliotchaim, thanks for reporting this issue and sorry for the late reply.

So far, your project config looks good to me and I'm not able to reproduce the issue.
To help us understand the issue a bit more, Could you send us the following information?
1. your function app sitename in Azure
2. your function core tools `func --version`

**Comment by AbdolrezaMadi** [comment]
Hello Guys,

I'm running into the same issue. The app is running fine locally, but when I publish it to Azure (West Europe) with --build-native-deps, I get the ModuleNotFoundError error (pandas or other modules). 

My function app name is rds-af-analytics-py
func --version is 2.4.419

requirements.txt:

[requirements.txt](https://github.com/Azure/azure-functions-python-worker/files/3552032/requirements.txt)

**Comment by anirudhgarg** [comment]
Can you folks please try func core tools publish with build --remote while publishing 

func azure functionapp publish <APP_NAME> --build remote

https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-function-python#deploy-the-function-app-project-to-azure

**Comment by LbrsRyanS** [comment]
Hi @anirudhgarg,

I am having the same issue as above.

Function App = 'project-peach'
func version = 2.7.1585

Could you advise how to deploy with the --build remote command through VS Code? Is it a PreDeployTask?

**Comment by AbdolrezaMadi** [comment]
Hi,
You can use --build remote with following syntax:
func azure functionapp publish <APP_NAME> --build remote

However, this is not solving my issue. Another error is popping up:

There was an error restoring dependencies.ERROR: ['pip', 'download', '-r', 'requirements.txt', '--dest', 'destination folder'] failed with exit code 1. 

Any suggestion?

**Comment by Hazhzeng** [comment]
@LbrsRyanS Please follow the vscode-azurefunctions wiki page for deploying with remote build https://github.com/microsoft/vscode-azurefunctions/wiki/Remote-Build

@AbdolrezaMadi When using the remote build, your package should be resolve on our servers instead of your local machine. Your functions core tools is out of date (2.4.419) and you can update it by `npm i -g azure-functions-core-tools --unsafe-perm true` and the **build --remote** flag should work for you afterwards.

**Comment by AbdolrezaMadi** [comment]
Thanks a lot for the tip. It solved the problem I mentioned. After fixing this issue, I got another problem with azure-function package, which is explained here:
https://github.com/Azure/azure-functions-python-worker/issues/504
Upgrdaing this package to version 1.0.4 and removing azure-functions-worker from my requirements.txt fixed the issues (I recreated the function app, since it was recommended in the Remote-Build procedure).

Thanks again
